docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+### GetResource service
+
+The GetResource grpc service interface provided by the attestation agent is used to obtain various confidential resources from the relying party (KBS). In the process of protected image deployment, it is usually used to obtain policiy.json and sigstore config file. 
+
+Attestation-agent performs HW-TEE attestation to the KBS, establish an encrypted channel after negotiating the key, and download the required confidential resources through the encrypted channel. The GetResource grpc service interface protobuf of the attachment agent is defined as follows:
+
+```protobuf
+message getResourceRequest {
+    string KbcName = 1;
+    string KbsUri = 2;
+    string ResourceDescription = 3;
+}
+
+message getResourceResponse {
+    bytes resource = 1;
+}
+
+service GetResource {
+    rpc GetResource(getResourceRequest) returns (getResourceResponse) {};
+}
+```
+
+ResourceDescription is a JSON string used to describe resources, and its format is defined as follows:
+
+```json
+{
+    "name":"example_resource",
+    "informations":{
+        "index":"example_index",
+        "extra_param":[]
+    },
+}
+```
+
+In the design of confidential containers V1, when get policy.json through GetResource grpc endpoint, the contents of ResourceDescription are as follows:
+
+```json
+{
+    "name":"policy.json",
+    "informations":{
+        "index":"",
+        "extra_param":[]
+    },
+}
+```
+
+When get sigstore config file through GetResource grpc endpoint, the contents of ResourceDescription are as follows:
+
+```json
+{
+    "name":"sigstore_config",
+    "informations":{
+        "index":"",
+        "extra_param":[]
+    },
+}
+```
+
+### KeyProvider service
+
+The keyprovider grpc interface provided by the attestation-agent is used to decrypt `PrivateLayerBlockCipherOptions` in cooperation with ocicrypt rs. Protobuf is defined as follows:
+
+```protobuf
+message keyProviderKeyWrapProtocolInput {
+    bytes KeyProviderKeyWrapProtocolInput = 1;
+}
+
+message keyProviderKeyWrapProtocolOutput {
+    bytes KeyProviderKeyWrapProtocolOutput = 1;
+}
+
+service KeyProviderService {
+    rpc UnWrapKey(keyProviderKeyWrapProtocolInput) returns (keyProviderKeyWrapProtocolOutput) {};
+}
+```
+
+The input JSON string format is:
+
+```json
+{
+    "op":"keyunwrap",
+    "keyunwrapparams":{
+        "dc":{
+            "Parameters":{
+                "attestation-agent":[
+                    "KBC_NAME::KBS_URI <base64encode>"
+                ],
+                "DecryptConfig":{"Parameters":{}}
+            }
+        }
+    },
+    "annotation": #annotation-packet,
+}
+```
+
+The output JSON string format is:
+
+```json
+{
+    "keyunwrapresults": {
+        "optsdata": #PrivateLayerBlockCipherOptions,
```

Comment:
Do you wrap this in Base64 or something?

Reply:
`optsdata` is a bit of a weird name. What about `PrivateLayerBlockCipherOption` for clarity?

Reply:
The data here is directly transmitted in the form of byte array without Base64 encoding.
The `optsdata` name is defined by the key provider protocol. I think we may should follow it.